docs/csharp/advanced-topics/interop/snippets/OfficeWalkthrough/ThisAddIn.cs

```diff
@@ -72,6 +106,109 @@ void DisplayInExcel(IEnumerable<Account> accounts,
         }
         //</Display>
 
+        //<ProperCleanup>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void CreateComObjectsAndCleanup()
+        {
+            Excel.Application excelApp = null;
+            Excel.Workbook workbook = null;
+            Excel.Worksheet worksheet = null;
+            
+            try
+            {
+                excelApp = new Excel.Application();
+                workbook = excelApp.Workbooks.Add();
+                worksheet = workbook.ActiveSheet;
+                
+                // Use COM objects here...
+            }
+            finally
+            {
+                // Clean up COM objects in reverse order of creation
+                if (worksheet != null)
+                {
+                    Marshal.FinalReleaseComObject(worksheet);
+                }
+                if (workbook != null)
+                {
+                    workbook.Close(true);
+                    Marshal.FinalReleaseComObject(workbook);
+                }
+                if (excelApp != null)
+                {
+                    excelApp.Quit();
```

Comment:
The other two samples have `excelApp.DisplayAlerts = true;` before this line. Is this sample missing it? Or should the line be dropped in the other samples?